// @lukso/lsp7-contracts/dist/abi.mjs
const ilsp7DigitalAssetAbi = [
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorAuthorizationChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "notified", internalType: "bool", type: "bool", indexed: true },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorRevoked"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "from", internalType: "address", type: "address", indexed: true },
      { name: "to", internalType: "address", type: "address", indexed: true },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      { name: "force", internalType: "bool", type: "bool", indexed: false },
      { name: "data", internalType: "bytes", type: "bytes", indexed: false }
    ],
    name: "Transfer"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "authorizeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" }
    ],
    name: "authorizedAmountFor",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "balanceOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "data", internalType: "bytes[]", type: "bytes[]" }],
    name: "batchCalls",
    outputs: [{ name: "results", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "decimals",
    outputs: [{ name: "", internalType: "uint8", type: "uint8" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "subtractedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "decreaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "getOperatorsOf",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "addedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "increaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "notify", internalType: "bool", type: "bool" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "revokeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "totalSupply",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "to", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "force", internalType: "bool", type: "bool" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "transfer",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address[]", type: "address[]" },
      { name: "to", internalType: "address[]", type: "address[]" },
      { name: "amount", internalType: "uint256[]", type: "uint256[]" },
      { name: "force", internalType: "bool[]", type: "bool[]" },
      { name: "data", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "transferBatch",
    outputs: [],
    stateMutability: "nonpayable"
  }
];
const lsp7BurnableAbi = [
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesEmptyArray" },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  { type: "error", inputs: [], name: "ERC725Y_MsgValueDisallowed" },
  {
    type: "error",
    inputs: [{ name: "storedData", internalType: "bytes", type: "bytes" }],
    name: "InvalidExtensionAddress"
  },
  {
    type: "error",
    inputs: [{ name: "data", internalType: "bytes", type: "bytes" }],
    name: "InvalidFunctionSelector"
  },
  { type: "error", inputs: [], name: "LSP4TokenNameNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenSymbolNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenTypeNotEditable" },
  {
    type: "error",
    inputs: [
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "authorizedAmount", internalType: "uint256", type: "uint256" },
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsAuthorizedAmount"
  },
  {
    type: "error",
    inputs: [
      { name: "balance", internalType: "uint256", type: "uint256" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsBalance"
  },
  {
    type: "error",
    inputs: [{ name: "callIndex", internalType: "uint256", type: "uint256" }],
    name: "LSP7BatchCallFailed"
  },
  { type: "error", inputs: [], name: "LSP7CannotSendWithAddressZero" },
  { type: "error", inputs: [], name: "LSP7CannotUseAddressZeroAsOperator" },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7DecreaseAllowanceNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7DecreasedAllowanceBelowZero" },
  { type: "error", inputs: [], name: "LSP7InvalidTransferBatch" },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverContractMissingLSP1Interface"
  },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverIsEOA"
  },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7RevokeOperatorNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7TokenContractCannotHoldValue" },
  { type: "error", inputs: [], name: "LSP7TokenOwnerCannotBeOperator" },
  {
    type: "error",
    inputs: [
      { name: "functionSelector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "NoExtensionFoundForFunctionSelector"
  },
  {
    type: "error",
    inputs: [{ name: "operator", internalType: "address", type: "address" }],
    name: "OperatorAllowanceCannotBeIncreasedFromZero"
  },
  { type: "error", inputs: [], name: "OwnableCannotSetZeroAddressAsOwner" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "dataKey",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "dataValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "DataChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorAuthorizationChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "notified", internalType: "bool", type: "bool", indexed: true },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorRevoked"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "from", internalType: "address", type: "address", indexed: true },
      { name: "to", internalType: "address", type: "address", indexed: true },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      { name: "force", internalType: "bool", type: "bool", indexed: false },
      { name: "data", internalType: "bytes", type: "bytes", indexed: false }
    ],
    name: "Transfer"
  },
  { type: "fallback", stateMutability: "payable" },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "authorizeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" }
    ],
    name: "authorizedAmountFor",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "balanceOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "data", internalType: "bytes[]", type: "bytes[]" }],
    name: "batchCalls",
    outputs: [{ name: "results", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "burn",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "decimals",
    outputs: [{ name: "", internalType: "uint8", type: "uint8" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "subtractedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "decreaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "getData",
    outputs: [{ name: "dataValue", internalType: "bytes", type: "bytes" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" }
    ],
    name: "getDataBatch",
    outputs: [{ name: "dataValues", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "getOperatorsOf",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "addedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "increaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "notify", internalType: "bool", type: "bool" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "revokeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "setData",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" },
      { name: "dataValues", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "setDataBatch",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "totalSupply",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "to", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "force", internalType: "bool", type: "bool" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "transfer",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address[]", type: "address[]" },
      { name: "to", internalType: "address[]", type: "address[]" },
      { name: "amount", internalType: "uint256[]", type: "uint256[]" },
      { name: "force", internalType: "bool[]", type: "bool[]" },
      { name: "data", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "transferBatch",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  { type: "receive", stateMutability: "payable" }
];
const lsp7BurnableInitAbstractAbi = [
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesEmptyArray" },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  { type: "error", inputs: [], name: "ERC725Y_MsgValueDisallowed" },
  {
    type: "error",
    inputs: [{ name: "storedData", internalType: "bytes", type: "bytes" }],
    name: "InvalidExtensionAddress"
  },
  {
    type: "error",
    inputs: [{ name: "data", internalType: "bytes", type: "bytes" }],
    name: "InvalidFunctionSelector"
  },
  { type: "error", inputs: [], name: "LSP4TokenNameNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenSymbolNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenTypeNotEditable" },
  {
    type: "error",
    inputs: [
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "authorizedAmount", internalType: "uint256", type: "uint256" },
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsAuthorizedAmount"
  },
  {
    type: "error",
    inputs: [
      { name: "balance", internalType: "uint256", type: "uint256" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsBalance"
  },
  {
    type: "error",
    inputs: [{ name: "callIndex", internalType: "uint256", type: "uint256" }],
    name: "LSP7BatchCallFailed"
  },
  { type: "error", inputs: [], name: "LSP7CannotSendWithAddressZero" },
  { type: "error", inputs: [], name: "LSP7CannotUseAddressZeroAsOperator" },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7DecreaseAllowanceNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7DecreasedAllowanceBelowZero" },
  { type: "error", inputs: [], name: "LSP7InvalidTransferBatch" },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverContractMissingLSP1Interface"
  },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverIsEOA"
  },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7RevokeOperatorNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7TokenContractCannotHoldValue" },
  { type: "error", inputs: [], name: "LSP7TokenOwnerCannotBeOperator" },
  {
    type: "error",
    inputs: [
      { name: "functionSelector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "NoExtensionFoundForFunctionSelector"
  },
  {
    type: "error",
    inputs: [{ name: "operator", internalType: "address", type: "address" }],
    name: "OperatorAllowanceCannotBeIncreasedFromZero"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "dataKey",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "dataValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "DataChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      { name: "version", internalType: "uint8", type: "uint8", indexed: false }
    ],
    name: "Initialized"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorAuthorizationChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "notified", internalType: "bool", type: "bool", indexed: true },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorRevoked"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "from", internalType: "address", type: "address", indexed: true },
      { name: "to", internalType: "address", type: "address", indexed: true },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      { name: "force", internalType: "bool", type: "bool", indexed: false },
      { name: "data", internalType: "bytes", type: "bytes", indexed: false }
    ],
    name: "Transfer"
  },
  { type: "fallback", stateMutability: "payable" },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "authorizeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" }
    ],
    name: "authorizedAmountFor",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "balanceOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "data", internalType: "bytes[]", type: "bytes[]" }],
    name: "batchCalls",
    outputs: [{ name: "results", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "burn",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "decimals",
    outputs: [{ name: "", internalType: "uint8", type: "uint8" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "subtractedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "decreaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "getData",
    outputs: [{ name: "dataValue", internalType: "bytes", type: "bytes" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" }
    ],
    name: "getDataBatch",
    outputs: [{ name: "dataValues", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "getOperatorsOf",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "addedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "increaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "notify", internalType: "bool", type: "bool" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "revokeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "setData",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" },
      { name: "dataValues", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "setDataBatch",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "totalSupply",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "to", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "force", internalType: "bool", type: "bool" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "transfer",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address[]", type: "address[]" },
      { name: "to", internalType: "address[]", type: "address[]" },
      { name: "amount", internalType: "uint256[]", type: "uint256[]" },
      { name: "force", internalType: "bool[]", type: "bool[]" },
      { name: "data", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "transferBatch",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  { type: "receive", stateMutability: "payable" }
];
const lsp7CappedSupplyAbi = [
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesEmptyArray" },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  { type: "error", inputs: [], name: "ERC725Y_MsgValueDisallowed" },
  {
    type: "error",
    inputs: [{ name: "storedData", internalType: "bytes", type: "bytes" }],
    name: "InvalidExtensionAddress"
  },
  {
    type: "error",
    inputs: [{ name: "data", internalType: "bytes", type: "bytes" }],
    name: "InvalidFunctionSelector"
  },
  { type: "error", inputs: [], name: "LSP4TokenNameNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenSymbolNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenTypeNotEditable" },
  {
    type: "error",
    inputs: [
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "authorizedAmount", internalType: "uint256", type: "uint256" },
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsAuthorizedAmount"
  },
  {
    type: "error",
    inputs: [
      { name: "balance", internalType: "uint256", type: "uint256" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsBalance"
  },
  {
    type: "error",
    inputs: [{ name: "callIndex", internalType: "uint256", type: "uint256" }],
    name: "LSP7BatchCallFailed"
  },
  { type: "error", inputs: [], name: "LSP7CannotSendWithAddressZero" },
  { type: "error", inputs: [], name: "LSP7CannotUseAddressZeroAsOperator" },
  { type: "error", inputs: [], name: "LSP7CappedSupplyCannotMintOverCap" },
  { type: "error", inputs: [], name: "LSP7CappedSupplyRequired" },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7DecreaseAllowanceNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7DecreasedAllowanceBelowZero" },
  { type: "error", inputs: [], name: "LSP7InvalidTransferBatch" },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverContractMissingLSP1Interface"
  },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverIsEOA"
  },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7RevokeOperatorNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7TokenContractCannotHoldValue" },
  { type: "error", inputs: [], name: "LSP7TokenOwnerCannotBeOperator" },
  {
    type: "error",
    inputs: [
      { name: "functionSelector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "NoExtensionFoundForFunctionSelector"
  },
  {
    type: "error",
    inputs: [{ name: "operator", internalType: "address", type: "address" }],
    name: "OperatorAllowanceCannotBeIncreasedFromZero"
  },
  { type: "error", inputs: [], name: "OwnableCannotSetZeroAddressAsOwner" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "dataKey",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "dataValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "DataChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorAuthorizationChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "notified", internalType: "bool", type: "bool", indexed: true },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorRevoked"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "from", internalType: "address", type: "address", indexed: true },
      { name: "to", internalType: "address", type: "address", indexed: true },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      { name: "force", internalType: "bool", type: "bool", indexed: false },
      { name: "data", internalType: "bytes", type: "bytes", indexed: false }
    ],
    name: "Transfer"
  },
  { type: "fallback", stateMutability: "payable" },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "authorizeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" }
    ],
    name: "authorizedAmountFor",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "balanceOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "data", internalType: "bytes[]", type: "bytes[]" }],
    name: "batchCalls",
    outputs: [{ name: "results", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "decimals",
    outputs: [{ name: "", internalType: "uint8", type: "uint8" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "subtractedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "decreaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "getData",
    outputs: [{ name: "dataValue", internalType: "bytes", type: "bytes" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" }
    ],
    name: "getDataBatch",
    outputs: [{ name: "dataValues", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "getOperatorsOf",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "addedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "increaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "notify", internalType: "bool", type: "bool" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "revokeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "setData",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" },
      { name: "dataValues", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "setDataBatch",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "tokenSupplyCap",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "totalSupply",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "to", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "force", internalType: "bool", type: "bool" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "transfer",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address[]", type: "address[]" },
      { name: "to", internalType: "address[]", type: "address[]" },
      { name: "amount", internalType: "uint256[]", type: "uint256[]" },
      { name: "force", internalType: "bool[]", type: "bool[]" },
      { name: "data", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "transferBatch",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  { type: "receive", stateMutability: "payable" }
];
const lsp7CappedSupplyInitAbstractAbi = [
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesEmptyArray" },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  { type: "error", inputs: [], name: "ERC725Y_MsgValueDisallowed" },
  {
    type: "error",
    inputs: [{ name: "storedData", internalType: "bytes", type: "bytes" }],
    name: "InvalidExtensionAddress"
  },
  {
    type: "error",
    inputs: [{ name: "data", internalType: "bytes", type: "bytes" }],
    name: "InvalidFunctionSelector"
  },
  { type: "error", inputs: [], name: "LSP4TokenNameNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenSymbolNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenTypeNotEditable" },
  {
    type: "error",
    inputs: [
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "authorizedAmount", internalType: "uint256", type: "uint256" },
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsAuthorizedAmount"
  },
  {
    type: "error",
    inputs: [
      { name: "balance", internalType: "uint256", type: "uint256" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsBalance"
  },
  {
    type: "error",
    inputs: [{ name: "callIndex", internalType: "uint256", type: "uint256" }],
    name: "LSP7BatchCallFailed"
  },
  { type: "error", inputs: [], name: "LSP7CannotSendWithAddressZero" },
  { type: "error", inputs: [], name: "LSP7CannotUseAddressZeroAsOperator" },
  { type: "error", inputs: [], name: "LSP7CappedSupplyCannotMintOverCap" },
  { type: "error", inputs: [], name: "LSP7CappedSupplyRequired" },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7DecreaseAllowanceNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7DecreasedAllowanceBelowZero" },
  { type: "error", inputs: [], name: "LSP7InvalidTransferBatch" },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverContractMissingLSP1Interface"
  },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverIsEOA"
  },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7RevokeOperatorNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7TokenContractCannotHoldValue" },
  { type: "error", inputs: [], name: "LSP7TokenOwnerCannotBeOperator" },
  {
    type: "error",
    inputs: [
      { name: "functionSelector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "NoExtensionFoundForFunctionSelector"
  },
  {
    type: "error",
    inputs: [{ name: "operator", internalType: "address", type: "address" }],
    name: "OperatorAllowanceCannotBeIncreasedFromZero"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "dataKey",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "dataValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "DataChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      { name: "version", internalType: "uint8", type: "uint8", indexed: false }
    ],
    name: "Initialized"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorAuthorizationChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "notified", internalType: "bool", type: "bool", indexed: true },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorRevoked"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "from", internalType: "address", type: "address", indexed: true },
      { name: "to", internalType: "address", type: "address", indexed: true },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      { name: "force", internalType: "bool", type: "bool", indexed: false },
      { name: "data", internalType: "bytes", type: "bytes", indexed: false }
    ],
    name: "Transfer"
  },
  { type: "fallback", stateMutability: "payable" },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "authorizeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" }
    ],
    name: "authorizedAmountFor",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "balanceOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "data", internalType: "bytes[]", type: "bytes[]" }],
    name: "batchCalls",
    outputs: [{ name: "results", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "decimals",
    outputs: [{ name: "", internalType: "uint8", type: "uint8" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "subtractedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "decreaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "getData",
    outputs: [{ name: "dataValue", internalType: "bytes", type: "bytes" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" }
    ],
    name: "getDataBatch",
    outputs: [{ name: "dataValues", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "getOperatorsOf",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "addedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "increaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "notify", internalType: "bool", type: "bool" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "revokeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "setData",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" },
      { name: "dataValues", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "setDataBatch",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "tokenSupplyCap",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "totalSupply",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "to", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "force", internalType: "bool", type: "bool" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "transfer",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address[]", type: "address[]" },
      { name: "to", internalType: "address[]", type: "address[]" },
      { name: "amount", internalType: "uint256[]", type: "uint256[]" },
      { name: "force", internalType: "bool[]", type: "bool[]" },
      { name: "data", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "transferBatch",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  { type: "receive", stateMutability: "payable" }
];
const lsp7DigitalAssetAbi = [
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesEmptyArray" },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  { type: "error", inputs: [], name: "ERC725Y_MsgValueDisallowed" },
  {
    type: "error",
    inputs: [{ name: "storedData", internalType: "bytes", type: "bytes" }],
    name: "InvalidExtensionAddress"
  },
  {
    type: "error",
    inputs: [{ name: "data", internalType: "bytes", type: "bytes" }],
    name: "InvalidFunctionSelector"
  },
  { type: "error", inputs: [], name: "LSP4TokenNameNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenSymbolNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenTypeNotEditable" },
  {
    type: "error",
    inputs: [
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "authorizedAmount", internalType: "uint256", type: "uint256" },
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsAuthorizedAmount"
  },
  {
    type: "error",
    inputs: [
      { name: "balance", internalType: "uint256", type: "uint256" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsBalance"
  },
  {
    type: "error",
    inputs: [{ name: "callIndex", internalType: "uint256", type: "uint256" }],
    name: "LSP7BatchCallFailed"
  },
  { type: "error", inputs: [], name: "LSP7CannotSendWithAddressZero" },
  { type: "error", inputs: [], name: "LSP7CannotUseAddressZeroAsOperator" },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7DecreaseAllowanceNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7DecreasedAllowanceBelowZero" },
  { type: "error", inputs: [], name: "LSP7InvalidTransferBatch" },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverContractMissingLSP1Interface"
  },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverIsEOA"
  },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7RevokeOperatorNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7TokenContractCannotHoldValue" },
  { type: "error", inputs: [], name: "LSP7TokenOwnerCannotBeOperator" },
  {
    type: "error",
    inputs: [
      { name: "functionSelector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "NoExtensionFoundForFunctionSelector"
  },
  {
    type: "error",
    inputs: [{ name: "operator", internalType: "address", type: "address" }],
    name: "OperatorAllowanceCannotBeIncreasedFromZero"
  },
  { type: "error", inputs: [], name: "OwnableCannotSetZeroAddressAsOwner" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "dataKey",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "dataValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "DataChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorAuthorizationChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "notified", internalType: "bool", type: "bool", indexed: true },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorRevoked"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "from", internalType: "address", type: "address", indexed: true },
      { name: "to", internalType: "address", type: "address", indexed: true },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      { name: "force", internalType: "bool", type: "bool", indexed: false },
      { name: "data", internalType: "bytes", type: "bytes", indexed: false }
    ],
    name: "Transfer"
  },
  { type: "fallback", stateMutability: "payable" },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "authorizeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" }
    ],
    name: "authorizedAmountFor",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "balanceOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "data", internalType: "bytes[]", type: "bytes[]" }],
    name: "batchCalls",
    outputs: [{ name: "results", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "decimals",
    outputs: [{ name: "", internalType: "uint8", type: "uint8" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "subtractedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "decreaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "getData",
    outputs: [{ name: "dataValue", internalType: "bytes", type: "bytes" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" }
    ],
    name: "getDataBatch",
    outputs: [{ name: "dataValues", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "getOperatorsOf",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "addedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "increaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "notify", internalType: "bool", type: "bool" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "revokeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "setData",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" },
      { name: "dataValues", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "setDataBatch",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "totalSupply",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "to", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "force", internalType: "bool", type: "bool" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "transfer",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address[]", type: "address[]" },
      { name: "to", internalType: "address[]", type: "address[]" },
      { name: "amount", internalType: "uint256[]", type: "uint256[]" },
      { name: "force", internalType: "bool[]", type: "bool[]" },
      { name: "data", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "transferBatch",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  { type: "receive", stateMutability: "payable" }
];
const lsp7DigitalAssetInitAbstractAbi = [
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesEmptyArray" },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  { type: "error", inputs: [], name: "ERC725Y_MsgValueDisallowed" },
  {
    type: "error",
    inputs: [{ name: "storedData", internalType: "bytes", type: "bytes" }],
    name: "InvalidExtensionAddress"
  },
  {
    type: "error",
    inputs: [{ name: "data", internalType: "bytes", type: "bytes" }],
    name: "InvalidFunctionSelector"
  },
  { type: "error", inputs: [], name: "LSP4TokenNameNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenSymbolNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenTypeNotEditable" },
  {
    type: "error",
    inputs: [
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "authorizedAmount", internalType: "uint256", type: "uint256" },
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsAuthorizedAmount"
  },
  {
    type: "error",
    inputs: [
      { name: "balance", internalType: "uint256", type: "uint256" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsBalance"
  },
  {
    type: "error",
    inputs: [{ name: "callIndex", internalType: "uint256", type: "uint256" }],
    name: "LSP7BatchCallFailed"
  },
  { type: "error", inputs: [], name: "LSP7CannotSendWithAddressZero" },
  { type: "error", inputs: [], name: "LSP7CannotUseAddressZeroAsOperator" },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7DecreaseAllowanceNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7DecreasedAllowanceBelowZero" },
  { type: "error", inputs: [], name: "LSP7InvalidTransferBatch" },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverContractMissingLSP1Interface"
  },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverIsEOA"
  },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7RevokeOperatorNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7TokenContractCannotHoldValue" },
  { type: "error", inputs: [], name: "LSP7TokenOwnerCannotBeOperator" },
  {
    type: "error",
    inputs: [
      { name: "functionSelector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "NoExtensionFoundForFunctionSelector"
  },
  {
    type: "error",
    inputs: [{ name: "operator", internalType: "address", type: "address" }],
    name: "OperatorAllowanceCannotBeIncreasedFromZero"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "dataKey",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "dataValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "DataChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      { name: "version", internalType: "uint8", type: "uint8", indexed: false }
    ],
    name: "Initialized"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorAuthorizationChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "notified", internalType: "bool", type: "bool", indexed: true },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorRevoked"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "from", internalType: "address", type: "address", indexed: true },
      { name: "to", internalType: "address", type: "address", indexed: true },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      { name: "force", internalType: "bool", type: "bool", indexed: false },
      { name: "data", internalType: "bytes", type: "bytes", indexed: false }
    ],
    name: "Transfer"
  },
  { type: "fallback", stateMutability: "payable" },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "authorizeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" }
    ],
    name: "authorizedAmountFor",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "balanceOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "data", internalType: "bytes[]", type: "bytes[]" }],
    name: "batchCalls",
    outputs: [{ name: "results", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "decimals",
    outputs: [{ name: "", internalType: "uint8", type: "uint8" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "subtractedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "decreaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "getData",
    outputs: [{ name: "dataValue", internalType: "bytes", type: "bytes" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" }
    ],
    name: "getDataBatch",
    outputs: [{ name: "dataValues", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "getOperatorsOf",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "addedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "increaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "notify", internalType: "bool", type: "bool" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "revokeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "setData",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" },
      { name: "dataValues", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "setDataBatch",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "totalSupply",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "to", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "force", internalType: "bool", type: "bool" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "transfer",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address[]", type: "address[]" },
      { name: "to", internalType: "address[]", type: "address[]" },
      { name: "amount", internalType: "uint256[]", type: "uint256[]" },
      { name: "force", internalType: "bool[]", type: "bool[]" },
      { name: "data", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "transferBatch",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  { type: "receive", stateMutability: "payable" }
];
const lsp7MintableAbi = [
  {
    type: "constructor",
    inputs: [
      { name: "name_", internalType: "string", type: "string" },
      { name: "symbol_", internalType: "string", type: "string" },
      { name: "newOwner_", internalType: "address", type: "address" },
      { name: "lsp4TokenType_", internalType: "uint256", type: "uint256" },
      { name: "isNonDivisible_", internalType: "bool", type: "bool" }
    ],
    stateMutability: "nonpayable"
  },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesEmptyArray" },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  { type: "error", inputs: [], name: "ERC725Y_MsgValueDisallowed" },
  {
    type: "error",
    inputs: [{ name: "storedData", internalType: "bytes", type: "bytes" }],
    name: "InvalidExtensionAddress"
  },
  {
    type: "error",
    inputs: [{ name: "data", internalType: "bytes", type: "bytes" }],
    name: "InvalidFunctionSelector"
  },
  { type: "error", inputs: [], name: "LSP4TokenNameNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenSymbolNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenTypeNotEditable" },
  {
    type: "error",
    inputs: [
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "authorizedAmount", internalType: "uint256", type: "uint256" },
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsAuthorizedAmount"
  },
  {
    type: "error",
    inputs: [
      { name: "balance", internalType: "uint256", type: "uint256" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsBalance"
  },
  {
    type: "error",
    inputs: [{ name: "callIndex", internalType: "uint256", type: "uint256" }],
    name: "LSP7BatchCallFailed"
  },
  { type: "error", inputs: [], name: "LSP7CannotSendWithAddressZero" },
  { type: "error", inputs: [], name: "LSP7CannotUseAddressZeroAsOperator" },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7DecreaseAllowanceNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7DecreasedAllowanceBelowZero" },
  { type: "error", inputs: [], name: "LSP7InvalidTransferBatch" },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverContractMissingLSP1Interface"
  },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverIsEOA"
  },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7RevokeOperatorNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7TokenContractCannotHoldValue" },
  { type: "error", inputs: [], name: "LSP7TokenOwnerCannotBeOperator" },
  {
    type: "error",
    inputs: [
      { name: "functionSelector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "NoExtensionFoundForFunctionSelector"
  },
  {
    type: "error",
    inputs: [{ name: "operator", internalType: "address", type: "address" }],
    name: "OperatorAllowanceCannotBeIncreasedFromZero"
  },
  { type: "error", inputs: [], name: "OwnableCannotSetZeroAddressAsOwner" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "dataKey",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "dataValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "DataChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorAuthorizationChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "notified", internalType: "bool", type: "bool", indexed: true },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorRevoked"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "from", internalType: "address", type: "address", indexed: true },
      { name: "to", internalType: "address", type: "address", indexed: true },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      { name: "force", internalType: "bool", type: "bool", indexed: false },
      { name: "data", internalType: "bytes", type: "bytes", indexed: false }
    ],
    name: "Transfer"
  },
  { type: "fallback", stateMutability: "payable" },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "authorizeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" }
    ],
    name: "authorizedAmountFor",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "balanceOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "data", internalType: "bytes[]", type: "bytes[]" }],
    name: "batchCalls",
    outputs: [{ name: "results", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "decimals",
    outputs: [{ name: "", internalType: "uint8", type: "uint8" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "subtractedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "decreaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "getData",
    outputs: [{ name: "dataValue", internalType: "bytes", type: "bytes" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" }
    ],
    name: "getDataBatch",
    outputs: [{ name: "dataValues", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "getOperatorsOf",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "addedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "increaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "to", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "force", internalType: "bool", type: "bool" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "mint",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "notify", internalType: "bool", type: "bool" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "revokeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "setData",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" },
      { name: "dataValues", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "setDataBatch",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "totalSupply",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "to", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "force", internalType: "bool", type: "bool" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "transfer",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address[]", type: "address[]" },
      { name: "to", internalType: "address[]", type: "address[]" },
      { name: "amount", internalType: "uint256[]", type: "uint256[]" },
      { name: "force", internalType: "bool[]", type: "bool[]" },
      { name: "data", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "transferBatch",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  { type: "receive", stateMutability: "payable" }
];
const lsp7MintableInitAbi = [
  { type: "constructor", inputs: [], stateMutability: "nonpayable" },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesEmptyArray" },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  { type: "error", inputs: [], name: "ERC725Y_MsgValueDisallowed" },
  {
    type: "error",
    inputs: [{ name: "storedData", internalType: "bytes", type: "bytes" }],
    name: "InvalidExtensionAddress"
  },
  {
    type: "error",
    inputs: [{ name: "data", internalType: "bytes", type: "bytes" }],
    name: "InvalidFunctionSelector"
  },
  { type: "error", inputs: [], name: "LSP4TokenNameNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenSymbolNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenTypeNotEditable" },
  {
    type: "error",
    inputs: [
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "authorizedAmount", internalType: "uint256", type: "uint256" },
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsAuthorizedAmount"
  },
  {
    type: "error",
    inputs: [
      { name: "balance", internalType: "uint256", type: "uint256" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsBalance"
  },
  {
    type: "error",
    inputs: [{ name: "callIndex", internalType: "uint256", type: "uint256" }],
    name: "LSP7BatchCallFailed"
  },
  { type: "error", inputs: [], name: "LSP7CannotSendWithAddressZero" },
  { type: "error", inputs: [], name: "LSP7CannotUseAddressZeroAsOperator" },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7DecreaseAllowanceNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7DecreasedAllowanceBelowZero" },
  { type: "error", inputs: [], name: "LSP7InvalidTransferBatch" },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverContractMissingLSP1Interface"
  },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverIsEOA"
  },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7RevokeOperatorNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7TokenContractCannotHoldValue" },
  { type: "error", inputs: [], name: "LSP7TokenOwnerCannotBeOperator" },
  {
    type: "error",
    inputs: [
      { name: "functionSelector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "NoExtensionFoundForFunctionSelector"
  },
  {
    type: "error",
    inputs: [{ name: "operator", internalType: "address", type: "address" }],
    name: "OperatorAllowanceCannotBeIncreasedFromZero"
  },
  { type: "error", inputs: [], name: "OwnableCannotSetZeroAddressAsOwner" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "dataKey",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "dataValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "DataChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      { name: "version", internalType: "uint8", type: "uint8", indexed: false }
    ],
    name: "Initialized"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorAuthorizationChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "notified", internalType: "bool", type: "bool", indexed: true },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorRevoked"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "from", internalType: "address", type: "address", indexed: true },
      { name: "to", internalType: "address", type: "address", indexed: true },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      { name: "force", internalType: "bool", type: "bool", indexed: false },
      { name: "data", internalType: "bytes", type: "bytes", indexed: false }
    ],
    name: "Transfer"
  },
  { type: "fallback", stateMutability: "payable" },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "authorizeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" }
    ],
    name: "authorizedAmountFor",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "balanceOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "data", internalType: "bytes[]", type: "bytes[]" }],
    name: "batchCalls",
    outputs: [{ name: "results", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "decimals",
    outputs: [{ name: "", internalType: "uint8", type: "uint8" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "subtractedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "decreaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "getData",
    outputs: [{ name: "dataValue", internalType: "bytes", type: "bytes" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" }
    ],
    name: "getDataBatch",
    outputs: [{ name: "dataValues", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "getOperatorsOf",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "addedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "increaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "name_", internalType: "string", type: "string" },
      { name: "symbol_", internalType: "string", type: "string" },
      { name: "newOwner_", internalType: "address", type: "address" },
      { name: "lsp4TokenType_", internalType: "uint256", type: "uint256" },
      { name: "isNonDivisible_", internalType: "bool", type: "bool" }
    ],
    name: "initialize",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "to", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "force", internalType: "bool", type: "bool" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "mint",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "notify", internalType: "bool", type: "bool" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "revokeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "setData",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" },
      { name: "dataValues", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "setDataBatch",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "totalSupply",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "to", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "force", internalType: "bool", type: "bool" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "transfer",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address[]", type: "address[]" },
      { name: "to", internalType: "address[]", type: "address[]" },
      { name: "amount", internalType: "uint256[]", type: "uint256[]" },
      { name: "force", internalType: "bool[]", type: "bool[]" },
      { name: "data", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "transferBatch",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  { type: "receive", stateMutability: "payable" }
];
const lsp7VotesAbi = [
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesEmptyArray" },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  { type: "error", inputs: [], name: "ERC725Y_MsgValueDisallowed" },
  {
    type: "error",
    inputs: [{ name: "storedData", internalType: "bytes", type: "bytes" }],
    name: "InvalidExtensionAddress"
  },
  {
    type: "error",
    inputs: [{ name: "data", internalType: "bytes", type: "bytes" }],
    name: "InvalidFunctionSelector"
  },
  { type: "error", inputs: [], name: "InvalidShortString" },
  { type: "error", inputs: [], name: "LSP4TokenNameNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenSymbolNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenTypeNotEditable" },
  {
    type: "error",
    inputs: [
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "authorizedAmount", internalType: "uint256", type: "uint256" },
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsAuthorizedAmount"
  },
  {
    type: "error",
    inputs: [
      { name: "balance", internalType: "uint256", type: "uint256" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsBalance"
  },
  {
    type: "error",
    inputs: [{ name: "callIndex", internalType: "uint256", type: "uint256" }],
    name: "LSP7BatchCallFailed"
  },
  { type: "error", inputs: [], name: "LSP7CannotSendWithAddressZero" },
  { type: "error", inputs: [], name: "LSP7CannotUseAddressZeroAsOperator" },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7DecreaseAllowanceNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7DecreasedAllowanceBelowZero" },
  { type: "error", inputs: [], name: "LSP7InvalidTransferBatch" },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverContractMissingLSP1Interface"
  },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverIsEOA"
  },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7RevokeOperatorNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7TokenContractCannotHoldValue" },
  { type: "error", inputs: [], name: "LSP7TokenOwnerCannotBeOperator" },
  {
    type: "error",
    inputs: [
      { name: "functionSelector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "NoExtensionFoundForFunctionSelector"
  },
  {
    type: "error",
    inputs: [{ name: "operator", internalType: "address", type: "address" }],
    name: "OperatorAllowanceCannotBeIncreasedFromZero"
  },
  { type: "error", inputs: [], name: "OwnableCannotSetZeroAddressAsOwner" },
  {
    type: "error",
    inputs: [{ name: "str", internalType: "string", type: "string" }],
    name: "StringTooLong"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "dataKey",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "dataValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "DataChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "delegator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "fromDelegate",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "toDelegate",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "DelegateChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "delegate",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "previousBalance",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      {
        name: "newBalance",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      }
    ],
    name: "DelegateVotesChanged"
  },
  { type: "event", anonymous: false, inputs: [], name: "EIP712DomainChanged" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorAuthorizationChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "notified", internalType: "bool", type: "bool", indexed: true },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorRevoked"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "from", internalType: "address", type: "address", indexed: true },
      { name: "to", internalType: "address", type: "address", indexed: true },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      { name: "force", internalType: "bool", type: "bool", indexed: false },
      { name: "data", internalType: "bytes", type: "bytes", indexed: false }
    ],
    name: "Transfer"
  },
  { type: "fallback", stateMutability: "payable" },
  {
    type: "function",
    inputs: [],
    name: "CLOCK_MODE",
    outputs: [{ name: "", internalType: "string", type: "string" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "DOMAIN_SEPARATOR",
    outputs: [{ name: "", internalType: "bytes32", type: "bytes32" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "authorizeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" }
    ],
    name: "authorizedAmountFor",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "balanceOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "data", internalType: "bytes[]", type: "bytes[]" }],
    name: "batchCalls",
    outputs: [{ name: "results", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "pos", internalType: "uint32", type: "uint32" }
    ],
    name: "checkpoints",
    outputs: [
      {
        name: "",
        internalType: "struct LSP7Votes.Checkpoint",
        type: "tuple",
        components: [
          { name: "fromBlock", internalType: "uint32", type: "uint32" },
          { name: "votes", internalType: "uint224", type: "uint224" }
        ]
      }
    ],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "clock",
    outputs: [{ name: "", internalType: "uint48", type: "uint48" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "decimals",
    outputs: [{ name: "", internalType: "uint8", type: "uint8" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "subtractedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "decreaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "delegatee", internalType: "address", type: "address" }],
    name: "delegate",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "delegatee", internalType: "address", type: "address" },
      { name: "nonce", internalType: "uint256", type: "uint256" },
      { name: "expiry", internalType: "uint256", type: "uint256" },
      { name: "v", internalType: "uint8", type: "uint8" },
      { name: "r", internalType: "bytes32", type: "bytes32" },
      { name: "s", internalType: "bytes32", type: "bytes32" }
    ],
    name: "delegateBySig",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "delegates",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "eip712Domain",
    outputs: [
      { name: "fields", internalType: "bytes1", type: "bytes1" },
      { name: "name", internalType: "string", type: "string" },
      { name: "version", internalType: "string", type: "string" },
      { name: "chainId", internalType: "uint256", type: "uint256" },
      { name: "verifyingContract", internalType: "address", type: "address" },
      { name: "salt", internalType: "bytes32", type: "bytes32" },
      { name: "extensions", internalType: "uint256[]", type: "uint256[]" }
    ],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "getData",
    outputs: [{ name: "dataValue", internalType: "bytes", type: "bytes" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" }
    ],
    name: "getDataBatch",
    outputs: [{ name: "dataValues", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "getOperatorsOf",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "timepoint", internalType: "uint256", type: "uint256" }],
    name: "getPastTotalSupply",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "timepoint", internalType: "uint256", type: "uint256" }
    ],
    name: "getPastVotes",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "getVotes",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "addedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "increaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "owner", internalType: "address", type: "address" }],
    name: "nonces",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "numCheckpoints",
    outputs: [{ name: "", internalType: "uint32", type: "uint32" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "notify", internalType: "bool", type: "bool" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "revokeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "setData",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" },
      { name: "dataValues", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "setDataBatch",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "totalSupply",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "to", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "force", internalType: "bool", type: "bool" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "transfer",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address[]", type: "address[]" },
      { name: "to", internalType: "address[]", type: "address[]" },
      { name: "amount", internalType: "uint256[]", type: "uint256[]" },
      { name: "force", internalType: "bool[]", type: "bool[]" },
      { name: "data", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "transferBatch",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  { type: "receive", stateMutability: "payable" }
];
const lsp7VotesInitAbstractAbi = [
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesEmptyArray" },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  { type: "error", inputs: [], name: "ERC725Y_MsgValueDisallowed" },
  {
    type: "error",
    inputs: [{ name: "storedData", internalType: "bytes", type: "bytes" }],
    name: "InvalidExtensionAddress"
  },
  {
    type: "error",
    inputs: [{ name: "data", internalType: "bytes", type: "bytes" }],
    name: "InvalidFunctionSelector"
  },
  { type: "error", inputs: [], name: "LSP4TokenNameNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenSymbolNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenTypeNotEditable" },
  {
    type: "error",
    inputs: [
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "authorizedAmount", internalType: "uint256", type: "uint256" },
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsAuthorizedAmount"
  },
  {
    type: "error",
    inputs: [
      { name: "balance", internalType: "uint256", type: "uint256" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP7AmountExceedsBalance"
  },
  {
    type: "error",
    inputs: [{ name: "callIndex", internalType: "uint256", type: "uint256" }],
    name: "LSP7BatchCallFailed"
  },
  { type: "error", inputs: [], name: "LSP7CannotSendWithAddressZero" },
  { type: "error", inputs: [], name: "LSP7CannotUseAddressZeroAsOperator" },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7DecreaseAllowanceNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7DecreasedAllowanceBelowZero" },
  { type: "error", inputs: [], name: "LSP7InvalidTransferBatch" },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverContractMissingLSP1Interface"
  },
  {
    type: "error",
    inputs: [
      { name: "tokenReceiver", internalType: "address", type: "address" }
    ],
    name: "LSP7NotifyTokenReceiverIsEOA"
  },
  {
    type: "error",
    inputs: [
      { name: "caller", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "operator", internalType: "address", type: "address" }
    ],
    name: "LSP7RevokeOperatorNotAuthorized"
  },
  { type: "error", inputs: [], name: "LSP7TokenContractCannotHoldValue" },
  { type: "error", inputs: [], name: "LSP7TokenOwnerCannotBeOperator" },
  {
    type: "error",
    inputs: [
      { name: "functionSelector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "NoExtensionFoundForFunctionSelector"
  },
  {
    type: "error",
    inputs: [{ name: "operator", internalType: "address", type: "address" }],
    name: "OperatorAllowanceCannotBeIncreasedFromZero"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "dataKey",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "dataValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "DataChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "delegator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "fromDelegate",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "toDelegate",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "DelegateChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "delegate",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "previousBalance",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      {
        name: "newBalance",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      }
    ],
    name: "DelegateVotesChanged"
  },
  { type: "event", anonymous: false, inputs: [], name: "EIP712DomainChanged" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      { name: "version", internalType: "uint8", type: "uint8", indexed: false }
    ],
    name: "Initialized"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorAuthorizationChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "tokenOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "notified", internalType: "bool", type: "bool", indexed: true },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "OperatorRevoked"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "operator",
        internalType: "address",
        type: "address",
        indexed: true
      },
      { name: "from", internalType: "address", type: "address", indexed: true },
      { name: "to", internalType: "address", type: "address", indexed: true },
      {
        name: "amount",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      { name: "force", internalType: "bool", type: "bool", indexed: false },
      { name: "data", internalType: "bytes", type: "bytes", indexed: false }
    ],
    name: "Transfer"
  },
  { type: "fallback", stateMutability: "payable" },
  {
    type: "function",
    inputs: [],
    name: "CLOCK_MODE",
    outputs: [{ name: "", internalType: "string", type: "string" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "DOMAIN_SEPARATOR",
    outputs: [{ name: "", internalType: "bytes32", type: "bytes32" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "authorizeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" }
    ],
    name: "authorizedAmountFor",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "balanceOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "data", internalType: "bytes[]", type: "bytes[]" }],
    name: "batchCalls",
    outputs: [{ name: "results", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "pos", internalType: "uint32", type: "uint32" }
    ],
    name: "checkpoints",
    outputs: [
      {
        name: "",
        internalType: "struct LSP7VotesInitAbstract.Checkpoint",
        type: "tuple",
        components: [
          { name: "fromBlock", internalType: "uint32", type: "uint32" },
          { name: "votes", internalType: "uint224", type: "uint224" }
        ]
      }
    ],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "clock",
    outputs: [{ name: "", internalType: "uint48", type: "uint48" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "decimals",
    outputs: [{ name: "", internalType: "uint8", type: "uint8" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "subtractedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "decreaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "delegatee", internalType: "address", type: "address" }],
    name: "delegate",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "delegatee", internalType: "address", type: "address" },
      { name: "nonce", internalType: "uint256", type: "uint256" },
      { name: "expiry", internalType: "uint256", type: "uint256" },
      { name: "v", internalType: "uint8", type: "uint8" },
      { name: "r", internalType: "bytes32", type: "bytes32" },
      { name: "s", internalType: "bytes32", type: "bytes32" }
    ],
    name: "delegateBySig",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "delegates",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "eip712Domain",
    outputs: [
      { name: "fields", internalType: "bytes1", type: "bytes1" },
      { name: "name", internalType: "string", type: "string" },
      { name: "version", internalType: "string", type: "string" },
      { name: "chainId", internalType: "uint256", type: "uint256" },
      { name: "verifyingContract", internalType: "address", type: "address" },
      { name: "salt", internalType: "bytes32", type: "bytes32" },
      { name: "extensions", internalType: "uint256[]", type: "uint256[]" }
    ],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "getData",
    outputs: [{ name: "dataValue", internalType: "bytes", type: "bytes" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" }
    ],
    name: "getDataBatch",
    outputs: [{ name: "dataValues", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "tokenOwner", internalType: "address", type: "address" }],
    name: "getOperatorsOf",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "timepoint", internalType: "uint256", type: "uint256" }],
    name: "getPastTotalSupply",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "timepoint", internalType: "uint256", type: "uint256" }
    ],
    name: "getPastVotes",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "getVotes",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "addedAmount", internalType: "uint256", type: "uint256" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "increaseAllowance",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "owner", internalType: "address", type: "address" }],
    name: "nonces",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "numCheckpoints",
    outputs: [{ name: "", internalType: "uint32", type: "uint32" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "operator", internalType: "address", type: "address" },
      { name: "tokenOwner", internalType: "address", type: "address" },
      { name: "notify", internalType: "bool", type: "bool" },
      {
        name: "operatorNotificationData",
        internalType: "bytes",
        type: "bytes"
      }
    ],
    name: "revokeOperator",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "setData",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" },
      { name: "dataValues", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "setDataBatch",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "totalSupply",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "to", internalType: "address", type: "address" },
      { name: "amount", internalType: "uint256", type: "uint256" },
      { name: "force", internalType: "bool", type: "bool" },
      { name: "data", internalType: "bytes", type: "bytes" }
    ],
    name: "transfer",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address[]", type: "address[]" },
      { name: "to", internalType: "address[]", type: "address[]" },
      { name: "amount", internalType: "uint256[]", type: "uint256[]" },
      { name: "force", internalType: "bool[]", type: "bool[]" },
      { name: "data", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "transferBatch",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  { type: "receive", stateMutability: "payable" }
];

export { ilsp7DigitalAssetAbi, lsp7BurnableAbi, lsp7BurnableInitAbstractAbi, lsp7CappedSupplyAbi, lsp7CappedSupplyInitAbstractAbi, lsp7DigitalAssetAbi, lsp7DigitalAssetInitAbstractAbi, lsp7MintableAbi, lsp7MintableInitAbi, lsp7VotesAbi, lsp7VotesInitAbstractAbi };
