// @lukso/lsp26-contracts/dist/abi.cjs
'use strict';

const ilsp26FollowerSystemAbi = [
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "follower",
        internalType: "address",
        type: "address",
        indexed: false
      },
      {
        name: "addr",
        internalType: "address",
        type: "address",
        indexed: false
      }
    ],
    name: "Follow"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "unfollower",
        internalType: "address",
        type: "address",
        indexed: false
      },
      {
        name: "addr",
        internalType: "address",
        type: "address",
        indexed: false
      }
    ],
    name: "Unfollow"
  },
  {
    type: "function",
    inputs: [{ name: "addr", internalType: "address", type: "address" }],
    name: "follow",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "addresses", internalType: "address[]", type: "address[]" }
    ],
    name: "followBatch",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "addr", internalType: "address", type: "address" }],
    name: "followerCount",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "addr", internalType: "address", type: "address" }],
    name: "followingCount",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "addr", internalType: "address", type: "address" },
      { name: "startIndex", internalType: "uint256", type: "uint256" },
      { name: "endIndex", internalType: "uint256", type: "uint256" }
    ],
    name: "getFollowersByIndex",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "addr", internalType: "address", type: "address" },
      { name: "startIndex", internalType: "uint256", type: "uint256" },
      { name: "endIndex", internalType: "uint256", type: "uint256" }
    ],
    name: "getFollowsByIndex",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "follower", internalType: "address", type: "address" },
      { name: "addr", internalType: "address", type: "address" }
    ],
    name: "isFollowing",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "addr", internalType: "address", type: "address" }],
    name: "unfollow",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "addresses", internalType: "address[]", type: "address[]" }
    ],
    name: "unfollowBatch",
    outputs: [],
    stateMutability: "nonpayable"
  }
];
const lsp26FollowerSystemAbi = [
  {
    type: "error",
    inputs: [{ name: "addr", internalType: "address", type: "address" }],
    name: "LSP26AlreadyFollowing"
  },
  { type: "error", inputs: [], name: "LSP26CannotSelfFollow" },
  {
    type: "error",
    inputs: [{ name: "addr", internalType: "address", type: "address" }],
    name: "LSP26NotFollowing"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "follower",
        internalType: "address",
        type: "address",
        indexed: false
      },
      {
        name: "addr",
        internalType: "address",
        type: "address",
        indexed: false
      }
    ],
    name: "Follow"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "unfollower",
        internalType: "address",
        type: "address",
        indexed: false
      },
      {
        name: "addr",
        internalType: "address",
        type: "address",
        indexed: false
      }
    ],
    name: "Unfollow"
  },
  {
    type: "function",
    inputs: [{ name: "addr", internalType: "address", type: "address" }],
    name: "follow",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "addresses", internalType: "address[]", type: "address[]" }
    ],
    name: "followBatch",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "addr", internalType: "address", type: "address" }],
    name: "followerCount",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "addr", internalType: "address", type: "address" }],
    name: "followingCount",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "addr", internalType: "address", type: "address" },
      { name: "startIndex", internalType: "uint256", type: "uint256" },
      { name: "endIndex", internalType: "uint256", type: "uint256" }
    ],
    name: "getFollowersByIndex",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "addr", internalType: "address", type: "address" },
      { name: "startIndex", internalType: "uint256", type: "uint256" },
      { name: "endIndex", internalType: "uint256", type: "uint256" }
    ],
    name: "getFollowsByIndex",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "follower", internalType: "address", type: "address" },
      { name: "addr", internalType: "address", type: "address" }
    ],
    name: "isFollowing",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "addr", internalType: "address", type: "address" }],
    name: "unfollow",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "addresses", internalType: "address[]", type: "address[]" }
    ],
    name: "unfollowBatch",
    outputs: [],
    stateMutability: "nonpayable"
  }
];

exports.ilsp26FollowerSystemAbi = ilsp26FollowerSystemAbi;
exports.lsp26FollowerSystemAbi = lsp26FollowerSystemAbi;
