// @lukso/lsp4-contracts/dist/abi.cjs
'use strict';

const lsp4DigitalAssetMetadataAbi = [
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesEmptyArray" },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  { type: "error", inputs: [], name: "ERC725Y_MsgValueDisallowed" },
  { type: "error", inputs: [], name: "LSP4TokenNameNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenSymbolNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenTypeNotEditable" },
  { type: "error", inputs: [], name: "OwnableCannotSetZeroAddressAsOwner" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "dataKey",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "dataValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "DataChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "function",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "getData",
    outputs: [{ name: "dataValue", internalType: "bytes", type: "bytes" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" }
    ],
    name: "getDataBatch",
    outputs: [{ name: "dataValues", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "setData",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" },
      { name: "dataValues", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "setDataBatch",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  }
];
const lsp4DigitalAssetMetadataInitAbstractAbi = [
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesEmptyArray" },
  { type: "error", inputs: [], name: "ERC725Y_DataKeysValuesLengthMismatch" },
  { type: "error", inputs: [], name: "ERC725Y_MsgValueDisallowed" },
  { type: "error", inputs: [], name: "LSP4TokenNameNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenSymbolNotEditable" },
  { type: "error", inputs: [], name: "LSP4TokenTypeNotEditable" },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "dataKey",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      },
      {
        name: "dataValue",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "DataChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      { name: "version", internalType: "uint8", type: "uint8", indexed: false }
    ],
    name: "Initialized"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "previousOwner",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "newOwner",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "OwnershipTransferred"
  },
  {
    type: "function",
    inputs: [{ name: "dataKey", internalType: "bytes32", type: "bytes32" }],
    name: "getData",
    outputs: [{ name: "dataValue", internalType: "bytes", type: "bytes" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" }
    ],
    name: "getDataBatch",
    outputs: [{ name: "dataValues", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "owner",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "renounceOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKey", internalType: "bytes32", type: "bytes32" },
      { name: "dataValue", internalType: "bytes", type: "bytes" }
    ],
    name: "setData",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "dataKeys", internalType: "bytes32[]", type: "bytes32[]" },
      { name: "dataValues", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "setDataBatch",
    outputs: [],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "newOwner", internalType: "address", type: "address" }],
    name: "transferOwnership",
    outputs: [],
    stateMutability: "nonpayable"
  }
];

exports.lsp4DigitalAssetMetadataAbi = lsp4DigitalAssetMetadataAbi;
exports.lsp4DigitalAssetMetadataInitAbstractAbi = lsp4DigitalAssetMetadataInitAbstractAbi;
