// @lukso/lsp11-contracts/dist/abi.cjs
'use strict';

const ilsp11SocialRecoveryAbi = [
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "guardian",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "GuardianAdded"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "guardian",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "GuardianRemoved"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "recoveryCounter",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      {
        name: "guardian",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "guardianVotedAddress",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "GuardianVotedFor"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "guardianThreshold",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      }
    ],
    name: "GuardiansThresholdChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "previousRecoveryCounter",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      }
    ],
    name: "RecoveryCancelled"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "recoveryDelay",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      }
    ],
    name: "RecoveryDelayChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "recoveryCounter",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "guardianVotedAddress",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "calldataExecuted",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "RecoveryProcessSuccessful"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "secretHash",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      }
    ],
    name: "SecretHashChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "recoveryCounter",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      {
        name: "committedBy",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "commitment",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      }
    ],
    name: "SecretHashCommitted"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "newGuardian", internalType: "address", type: "address" }
    ],
    name: "addGuardian",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "cancelRecoveryProcess",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "votedAddress", internalType: "address", type: "address" },
      { name: "commitment", internalType: "bytes32", type: "bytes32" }
    ],
    name: "commitToRecover",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "recoveryCounter", internalType: "uint256", type: "uint256" },
      { name: "committedBy", internalType: "address", type: "address" }
    ],
    name: "getCommitmentInfoOf",
    outputs: [
      { name: "", internalType: "bytes32", type: "bytes32" },
      { name: "", internalType: "uint256", type: "uint256" }
    ],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "recoveryCounter", internalType: "uint256", type: "uint256" }
    ],
    name: "getFirstRecoveryTimestampOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "getGuardiansOf",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "getGuardiansThresholdOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "getRecoveryCounterOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "getRecoveryDelayOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "getSecretHashOf",
    outputs: [{ name: "", internalType: "bytes32", type: "bytes32" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "recoveryCounter", internalType: "uint256", type: "uint256" },
      { name: "guardian", internalType: "address", type: "address" }
    ],
    name: "getVotedAddressByGuardian",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "recoveryCounter", internalType: "uint256", type: "uint256" },
      { name: "votedAddress", internalType: "address", type: "address" }
    ],
    name: "getVotesOfGuardianVotedAddress",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "recoveryCounter", internalType: "uint256", type: "uint256" },
      { name: "votedAddress", internalType: "address", type: "address" }
    ],
    name: "hasReachedThreshold",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "guardianAddress", internalType: "address", type: "address" }
    ],
    name: "isGuardianOf",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "votedAddress", internalType: "address", type: "address" },
      { name: "account", internalType: "address", type: "address" },
      { name: "secretHash", internalType: "bytes32", type: "bytes32" },
      { name: "newSecretHash", internalType: "bytes32", type: "bytes32" },
      { name: "calldataToExecute", internalType: "bytes", type: "bytes" }
    ],
    name: "recoverAccess",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "existingGuardian", internalType: "address", type: "address" }
    ],
    name: "removeGuardian",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "newThreshold", internalType: "uint256", type: "uint256" }
    ],
    name: "setGuardiansThreshold",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "recoveryDelay", internalType: "uint256", type: "uint256" }
    ],
    name: "setRecoveryDelay",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      {
        name: "newRecoverSecretHash",
        internalType: "bytes32",
        type: "bytes32"
      }
    ],
    name: "setRecoverySecretHash",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "guardian", internalType: "address", type: "address" },
      {
        name: "guardianVotedAddress",
        internalType: "address",
        type: "address"
      }
    ],
    name: "voteForRecovery",
    outputs: [],
    stateMutability: "nonpayable"
  }
];
const lsp11SocialRecoveryAbi = [
  { type: "error", inputs: [], name: "AccountNotSetupYet" },
  {
    type: "error",
    inputs: [{ name: "iteration", internalType: "uint256", type: "uint256" }],
    name: "BatchCallsFailed"
  },
  {
    type: "error",
    inputs: [],
    name: "BatchExecuteRelayCallParamsLengthMismatch"
  },
  {
    type: "error",
    inputs: [
      { name: "guardian", internalType: "address", type: "address" },
      { name: "caller", internalType: "address", type: "address" }
    ],
    name: "CallerIsNotGuardian"
  },
  {
    type: "error",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "caller", internalType: "address", type: "address" }
    ],
    name: "CallerIsNotTheAccount"
  },
  {
    type: "error",
    inputs: [
      { name: "votedAddress", internalType: "address", type: "address" },
      { name: "caller", internalType: "address", type: "address" }
    ],
    name: "CallerIsNotVotedAddress"
  },
  {
    type: "error",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "recoverer", internalType: "address", type: "address" }
    ],
    name: "CallerVotesHaveNotReachedThreshold"
  },
  {
    type: "error",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "committer", internalType: "address", type: "address" }
    ],
    name: "CannotRecoverAfterDirectCommit"
  },
  {
    type: "error",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "delay", internalType: "uint256", type: "uint256" }
    ],
    name: "CannotRecoverBeforeDelay"
  },
  {
    type: "error",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "guardian", internalType: "address", type: "address" },
      {
        name: "guardianVotedAddress",
        internalType: "address",
        type: "address"
      }
    ],
    name: "CannotVoteToAddressTwice"
  },
  {
    type: "error",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "guardian", internalType: "address", type: "address" }
    ],
    name: "GuardianAlreadyExists"
  },
  {
    type: "error",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "guardian", internalType: "address", type: "address" }
    ],
    name: "GuardianNotFound"
  },
  {
    type: "error",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "threshold", internalType: "uint256", type: "uint256" }
    ],
    name: "GuardianNumberCannotGoBelowThreshold"
  },
  {
    type: "error",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "committer", internalType: "address", type: "address" }
    ],
    name: "InvalidCommitment"
  },
  {
    type: "error",
    inputs: [
      { name: "signer", internalType: "address", type: "address" },
      { name: "invalidNonce", internalType: "uint256", type: "uint256" },
      { name: "signature", internalType: "bytes", type: "bytes" }
    ],
    name: "InvalidRelayNonce"
  },
  {
    type: "error",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "secretHash", internalType: "bytes32", type: "bytes32" }
    ],
    name: "InvalidSecretHash"
  },
  {
    type: "error",
    inputs: [
      { name: "totalValues", internalType: "uint256", type: "uint256" },
      { name: "msgValue", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP11BatchExcessiveValueSent"
  },
  {
    type: "error",
    inputs: [
      { name: "totalValues", internalType: "uint256", type: "uint256" },
      { name: "msgValue", internalType: "uint256", type: "uint256" }
    ],
    name: "LSP11BatchInsufficientValueSent"
  },
  {
    type: "error",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "nonGuardian", internalType: "address", type: "address" }
    ],
    name: "NotAGuardianOfTheAccount"
  },
  { type: "error", inputs: [], name: "RelayCallBeforeStartTime" },
  { type: "error", inputs: [], name: "RelayCallExpired" },
  {
    type: "error",
    inputs: [
      { name: "functionSelector", internalType: "bytes4", type: "bytes4" }
    ],
    name: "RelayCallNotSupported"
  },
  {
    type: "error",
    inputs: [
      { name: "votedAddress", internalType: "address", type: "address" },
      { name: "recoveredAddress", internalType: "address", type: "address" }
    ],
    name: "SignerIsNotVotedAddress"
  },
  {
    type: "error",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "threshold", internalType: "uint256", type: "uint256" }
    ],
    name: "ThresholdExceedsGuardianNumber"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "guardian",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "GuardianAdded"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "guardian",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "GuardianRemoved"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "recoveryCounter",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      {
        name: "guardian",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "guardianVotedAddress",
        internalType: "address",
        type: "address",
        indexed: true
      }
    ],
    name: "GuardianVotedFor"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "guardianThreshold",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      }
    ],
    name: "GuardiansThresholdChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "previousRecoveryCounter",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      }
    ],
    name: "RecoveryCancelled"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "recoveryDelay",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      }
    ],
    name: "RecoveryDelayChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "recoveryCounter",
        internalType: "uint256",
        type: "uint256",
        indexed: true
      },
      {
        name: "guardianVotedAddress",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "calldataExecuted",
        internalType: "bytes",
        type: "bytes",
        indexed: false
      }
    ],
    name: "RecoveryProcessSuccessful"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "secretHash",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      }
    ],
    name: "SecretHashChanged"
  },
  {
    type: "event",
    anonymous: false,
    inputs: [
      {
        name: "account",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "recoveryCounter",
        internalType: "uint256",
        type: "uint256",
        indexed: false
      },
      {
        name: "committedBy",
        internalType: "address",
        type: "address",
        indexed: true
      },
      {
        name: "commitment",
        internalType: "bytes32",
        type: "bytes32",
        indexed: true
      }
    ],
    name: "SecretHashCommitted"
  },
  {
    type: "function",
    inputs: [],
    name: "COMMITMEMT_DELAY",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [],
    name: "DEFAULT_RECOVERY_DELAY",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "newGuardian", internalType: "address", type: "address" }
    ],
    name: "addGuardian",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "data", internalType: "bytes[]", type: "bytes[]" }],
    name: "batchCalls",
    outputs: [{ name: "results", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "cancelRecoveryProcess",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "votedAddress", internalType: "address", type: "address" },
      { name: "commitment", internalType: "bytes32", type: "bytes32" }
    ],
    name: "commitToRecover",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "signature", internalType: "bytes", type: "bytes" },
      { name: "nonce", internalType: "uint256", type: "uint256" },
      { name: "validityTimestamps", internalType: "uint256", type: "uint256" },
      { name: "payload", internalType: "bytes", type: "bytes" }
    ],
    name: "executeRelayCall",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "signatures", internalType: "bytes[]", type: "bytes[]" },
      { name: "nonces", internalType: "uint256[]", type: "uint256[]" },
      {
        name: "validityTimestamps",
        internalType: "uint256[]",
        type: "uint256[]"
      },
      { name: "values", internalType: "uint256[]", type: "uint256[]" },
      { name: "payloads", internalType: "bytes[]", type: "bytes[]" }
    ],
    name: "executeRelayCallBatch",
    outputs: [{ name: "", internalType: "bytes[]", type: "bytes[]" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "recoveryCounter", internalType: "uint256", type: "uint256" },
      { name: "committedBy", internalType: "address", type: "address" }
    ],
    name: "getCommitmentInfoOf",
    outputs: [
      { name: "", internalType: "bytes32", type: "bytes32" },
      { name: "", internalType: "uint256", type: "uint256" }
    ],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "recoveryCounter", internalType: "uint256", type: "uint256" }
    ],
    name: "getFirstRecoveryTimestampOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "getGuardiansOf",
    outputs: [{ name: "", internalType: "address[]", type: "address[]" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "getGuardiansThresholdOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "from", internalType: "address", type: "address" },
      { name: "channelId", internalType: "uint128", type: "uint128" }
    ],
    name: "getNonce",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "getRecoveryCounterOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "getRecoveryDelayOf",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [{ name: "account", internalType: "address", type: "address" }],
    name: "getSecretHashOf",
    outputs: [{ name: "", internalType: "bytes32", type: "bytes32" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "recoveryCounter", internalType: "uint256", type: "uint256" },
      { name: "guardian", internalType: "address", type: "address" }
    ],
    name: "getVotedAddressByGuardian",
    outputs: [{ name: "", internalType: "address", type: "address" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "recoveryCounter", internalType: "uint256", type: "uint256" },
      { name: "votedAddress", internalType: "address", type: "address" }
    ],
    name: "getVotesOfGuardianVotedAddress",
    outputs: [{ name: "", internalType: "uint256", type: "uint256" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "recoveryCounter", internalType: "uint256", type: "uint256" },
      { name: "votedAddress", internalType: "address", type: "address" }
    ],
    name: "hasReachedThreshold",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "guardianAddress", internalType: "address", type: "address" }
    ],
    name: "isGuardianOf",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "votedAddress", internalType: "address", type: "address" },
      { name: "secretHash", internalType: "bytes32", type: "bytes32" },
      { name: "newSecretHash", internalType: "bytes32", type: "bytes32" },
      { name: "calldataToExecute", internalType: "bytes", type: "bytes" }
    ],
    name: "recoverAccess",
    outputs: [{ name: "", internalType: "bytes", type: "bytes" }],
    stateMutability: "payable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "existingGuardian", internalType: "address", type: "address" }
    ],
    name: "removeGuardian",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "newThreshold", internalType: "uint256", type: "uint256" }
    ],
    name: "setGuardiansThreshold",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "recoveryDelay", internalType: "uint256", type: "uint256" }
    ],
    name: "setRecoveryDelay",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      {
        name: "newRecoverSecretHash",
        internalType: "bytes32",
        type: "bytes32"
      }
    ],
    name: "setRecoverySecretHash",
    outputs: [],
    stateMutability: "nonpayable"
  },
  {
    type: "function",
    inputs: [{ name: "interfaceId", internalType: "bytes4", type: "bytes4" }],
    name: "supportsInterface",
    outputs: [{ name: "", internalType: "bool", type: "bool" }],
    stateMutability: "view"
  },
  {
    type: "function",
    inputs: [
      { name: "account", internalType: "address", type: "address" },
      { name: "guardian", internalType: "address", type: "address" },
      {
        name: "guardianVotedAddress",
        internalType: "address",
        type: "address"
      }
    ],
    name: "voteForRecovery",
    outputs: [],
    stateMutability: "nonpayable"
  }
];

exports.ilsp11SocialRecoveryAbi = ilsp11SocialRecoveryAbi;
exports.lsp11SocialRecoveryAbi = lsp11SocialRecoveryAbi;
